tests/Doctrine/Tests/DBAL/Functional/DataAccessTest.php

```diff
@@ -863,6 +863,25 @@ private function skipOci8AndMysqli()
             $this->markTestSkipped('Mysqli driver dont support this feature.');
         }
     }
+
+    public function testIteratorCanBeIteratedMultipleTimes()
+    {
+        $sql = "SELECT test_int, test_string FROM fetch_table";
+        $stmt = $this->_conn->query($sql);
+        $stmt->setFetchMode(\PDO::FETCH_ASSOC);
+
+        $i = 0;
+        foreach ($stmt as $row) {
+            $i++;
+        }
+
+        $j = 0;
+        foreach ($stmt as $row2) {
+            $j++;
+        }
+
+        $this->assertEquals($i, $j);
```

Comment:
afaict, travis is using mariadb which I think supports microseconds, so think it should be fine?

On another note, I noticed the build is failing and it's because pdo doesn't actually support rewinding the iterator (well not by default is my understanding, may be able to with unbuffered queries) so made me questions if we even needed to support this for non pdo drivers, or if we should, should we wrap the pdo statements so they can be rewound too?